tests/TestCase/Database/QueryTest.php

```diff
@@ -4697,4 +4669,75 @@ public function assertQuotedQuery($pattern, $query, $optional = false)
         $pattern = str_replace('>', '[`"\]]' . $optional, $pattern);
         $this->assertRegExp('#' . $pattern . '#', $query);
     }
+
+    /**
+     * Test that calling fetchAssoc return an associated array.
+     * @return void
+     * @throws \Exception
+     */
+    public function testFetchAssoc()
+    {
+        $this->loadFixtures('Profiles');
+        $query = new Query($this->connection);
+        $results = $query
+            ->select([
+                'id',
+                'user_id',
+                'is_active'
+            ])
+            ->from('profiles')
+            ->limit(1)
+            ->execute()
+            ->fetchAssoc();
+        $this->assertSame(['id' => '1', 'user_id' => '1', 'is_active' => '0'], $results);
+    }
+
+    /**
+     * Test that fetchColumn() will return the correct value at $position.
+     * @throws \Exception
+     * @return void
+     */
+    public function testFetchColumn()
+    {
+        $this->loadFixtures('Profiles');
+        $query = new Query($this->connection);
+        $query
+            ->select([
+                'id',
+                'user_id',
+                'is_active'
+            ])
+            ->from('profiles')
+            ->where(['id' => 2])
+            ->limit(1);
+        $statement = $query->execute();
+        $results = $statement->fetchColumn(0);
+        $this->assertEquals('2', $results[0]);
+    }
+
+    /**
+     * Test that calling fetchAssoc, fetchColum and fetchObject in sequence
+     * alters the fetched data to the correct types and values.
+     * @return void
+     * @throws \Exception
+     */
+    public function testFetchAllAssocColumn()
+    {
+        $this->loadFixtures('Profiles');
+        $query = new Query($this->connection);
+        $query
+            ->select([
+                'id',
+                'user_id',
+                'is_active'
+            ])
+            ->from('profiles');
+        $statement = $query->execute();
+        $results = $statement->fetchAssoc();
+        $this->assertEquals('1', $results['id']);
+        $results = $statement->fetchAssoc();
+        $this->assertEquals('2', $results['id']);
+        $results = $statement->fetchColumn(0);
+        $this->assertEquals('3', $results[0]);
```

Comment:
These should always be assertSame(), to avoid numeric 2 etc to also false positive here.

Reply:
Thanks for the tip.